omegaconf/dictconfig.py

```diff
@@ -574,6 +574,8 @@ def _set_value(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> Non
             self.__dict__["_content"] = "???"
             self._metadata.object_type = None
         else:
+            # shallow copy of content
```

Comment:
instead of sprinkling this function with error handling, you can do something like:
```python
def _set_value(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> None:
  try:
     # copy backup
     self._set_value_impl(...)
  except:
    # restore backup
    raise

def _set_value_impl(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> None:
   ... # same logic as the current function before your changes.
```

Reply:
Makes more sense.